### PR TITLE
feat(browser): pre-compile mongoose/browser

### LINKED ISSUE #7219: Pre-compile mongoose/browser
So people don't have to merge their webpack config with [Mongoose's recommended config](https://mongoosejs.com/docs/browser.html#webpack). Re: discussion from https://github.com/Automattic/mongoose/issues/6826#issuecomment-435933307

### PR DESCRIPTION
**Pull request by captaincaius**
Fix #7219

Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory.

If you're making a change to documentation, do **not** modify a `.html` file directly. Instead find the corresponding `.pug` file or test case in the `test/docs` directory.

**Summary**

Prebuilt mongoose/browser configured in a way that can be consumed in environments that don't have "window" too.  Webpack test also updated to test the exact same webpack config as the build.

**Examples**

### CODE REVIEW

**Comment by captaincaius** [comment]
Note: if you don't want a headache, ignore whitespace while reviewing this one ;)

File: package.json [hunk 1]
```diff
@@ -64,13 +64,16 @@
     "uuid": "2.0.3",
     "uuid-parse": "1.0.0",
     "validator": "10.8.0",
-    "webpack": "4.16.4"
+    "webpack": "4.16.4",
+    "webpack-cli": "^3.3.9"
```
Comment (vkarpov15) [inline package.json]: No up-carats please. Also, I'd really rather [run webpack from Node.js](https://masteringjs.io/tutorials/webpack/node) than pull in all of webpack-cli...

Comment (captaincaius) [reply package.json]: Ah ok.  I'll remove it entirely and make a little script instead.

Also, I seem to remember some alternative "light" CLI.  I'll check just how light it really is.

Comment (captaincaius) [reply package.json]: Never mind...  Looks like "webpack-command" has been deprecated.

Comment (captaincaius) [reply package.json]: Ready.  Removed the dependency all together.

**Comment by captaincaius** [comment]
Damn!  I merged in master to resolve the conflicts, not realizing I should have merged in 5.8 >:0.  I guess I need to redo this with cherry-pick, unless those other commits I pulled in will be going into 5.8 too.

**Comment by captaincaius** [comment]
undid the hell I caused here in #8350 so closing this one.

**Pull request closed without merging**